src/mds/MDSRank.cc

```diff
@@ -3571,9 +3573,11 @@ MDSRankDispatcher::MDSRankDispatcher(
     MgrClient *mgrc,
     Context *respawn_hook_,
     Context *suicide_hook_,
+    bool bal_export_pin,
     boost::asio::io_context& ioc)
   : MDSRank(whoami_, fs_name_, mds_lock_, clog_, timer_, beacon_, mdsmap_,
-            msgr, monc_, mgrc, respawn_hook_, suicide_hook_, ioc)
+            msgr, monc_, mgrc, respawn_hook_, suicide_hook_, 
+	    g_conf().get_val<bool>("mds_bal_export_pin"), ioc)
```

Comment:
@batrick PTAL here (and https://github.com/ceph/ceph/pull/35178/files#diff-ea4df8dd62d8f79137676b9f21d58633R490 and https://github.com/ceph/ceph/pull/35178/files#diff-89e22a97695f436e6645ce25f7bfb985R753) . It works but I'm not sure if it's proper hygiene.

Reply:
Why not just initialize it in the constructor body? I don't understand the reason for adding an argument to the constructor.

Reply:
Ack. Yea that makes more sense. Updated